src/tests/Common/Directory.Build.targets

```diff
@@ -9,6 +9,7 @@
           Condition="'$(DisableVersionCheckImported)' != 'true'" />
 
   <Import Project="$(RepoRoot)eng/liveBuilds.targets" />
+  <Import Project="$(RepositoryEngineeringDir)toolAot.targets" />
```

Comment:
Do we still need to build crossgen during test build? Would it make more sense to go back to using the one created during the build?

Reply:
Looks like there are still reasons to keep this. In particular, I moved all the package overrides here, so if we need to use new ILC packages the overrides are here.

Reply:
I am talking about https://github.com/dotnet/runtime/blob/main/src/tests/Common/Directory.Build.targets#L66-L78 and `UsePublishedCrossgen2` property. Building tests is reaching out into the product build to publish crossgen2. It should not be needed anymore since the ordering problems that this was working around should not exist after this PR. We should be able to use what we have produced during the product build instead. (My assumption was that the line that I have commented on is necessary to make this work.)

I agree that ability to override ILC LKG as necessary is good. But building tests should not need ILC LKG to publish. If tests need to publish, they should use the live ILC.

Reply:
Do you have thoughts about this? Should triggering build of crossgen from the test partition be deleted? (The pain associated with that actually started the conversation that led to this PR.)

Reply:
#108693